ansible_files/roles/aws/prepare_inventory/tasks/main.yml

```diff
@@ -0,0 +1,46 @@
+# tasks/main.yml
+
+- name: Load EC2 instances definitions from file
+  ansible.builtin.include_vars:
+    file: "{{ role_path }}/files/ec2_instances.yml"
```

Comment:
I think it would be more flexible if passed as an extra vars when running `ansible-playbook` (`ansible-playbook -e @path/to/var/file.yml`)